Gridfeedoc (#222)
BESTenergytrade/simply: doc/source/matching_strategies.rst

| Origin Cluster: 1  | 1                        | 0                       |
+--------------------+--------------------------+-------------------------+

.. _grid_fee_matrix:

Methods to Enter Grid Fee Matrix
--------------------------------

> There are two primary methods to input a grid fee matrix into the Simply Simulation:

Reviewer: ```suggestion
There are two methods to define a grid fee matrix for the simply simulation:
```

1. Providing a Complete Grid Fee Matrix

> 2. Building a Grid Fee Matrix from Configuration Files

Reviewer: maybe here reference links to the subsections below could be used

Providing an Already Complete Grid Fee Matrix
---------------------------------------------

Format
~~~~~~

- The grid fee matrix should be provided in a structured format such as JSON or CSV.

Placement
~~~~~~~~~

- Place the complete grid fee matrix in the designated input location within the simulation environment, as specified in the simulation documentation or configuration settings.

Requirements
~~~~~~~~~~~~

Names/Identifiers
"""""""""""""""""

- Each row and column must be labeled with unique identifiers representing different clusters or nodes.
- Example Identifiers:
  - **Network Lines:** line_1, line_2, line_n
  - **Nodes or Clusters:** node_A, node_B, node_C

Units
"""""

- Specify consistent units for both row and column headers to avoid confusion (e.g., USD, EUR).

Weight Factor
"""""""""""""

> - The weight factor describes the relationship between grid fee and cumulative power network edge weights.

Reviewer: ```suggestion
- The weight factor describes the relationship between grid fee and cumulative power network edge weight between two nodes in the network. The cumulative weight is the sum of all edge weights along the shortest path in the network between those nodes. If there are multiple nodes per cluster, nodes of one cluster have equivalent cumulative weights to all nodes of of another cluster.
```

> 

Reviewer: too many section, does not look very compact, maybe try
- `**Format**: The grid fee matrix should be provided in a structured format such as JSON or CSV.` etc.
- or as a two column table

Example JSON Grid Fee Matrix
^^^^^^^^^^^^^^^^^^^^^^^^^^^^

:: 

   {
     "grid_fees": [
       {
         "name": "transmission_fee",
         "unit": "$/kWh",
         "value": 0.05
       },
       {
         "name": "distribution_fee",
         "unit": "$/kWh",
         "value": 0.03
       }
     ]

>    }

Reviewer: Where can I find this in the code? Is it in another branch?

Author: I created this as an example.

Building a Grid Fee Matrix from Config Files
--------------------------------------------

Composition from Config Files

> """"""""""""""""""""""""""""""

Reviewer: not visible

The grid fee matrix can be composed by reading entries from configuration files using the following attributes:

> - ``default_grid_fee``: Default grid fee used by the market maker.
> - ``local_grid_fee``: Local grid fee to be used.
> - ``weight_factor``: Factor describing the relationship of grid fee to cumulative power network edge weights.

Reviewer: ```suggestion
- ``default_grid_fee``: Default grid fee applied for trades with the market maker
- ``local_grid_fee``: Local grid fee applied for trades within a cluster
- ``weight_factor``: Factor describing the relationship of grid fee to cumulative power network edge weights (see description above)
```

Steps to Build Grid Fee Matrix
""""""""""""""""""""""""""""""

1. Retrieve ``default_grid_fee``, ``local_grid_fee``, and ``weight_factor`` values from the configuration file.

> 2. Use these values to calculate or define the grid fee matrix entries.

Reviewer: This is happening within simply, steps do not need to be explained (i.e. can be removed), ... parameter are already explained above

Example Configuration Entries
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

:: 

   #--------------------------
   # market
   #--------------------------
   # default grid_fee to be used by market maker
   default_grid_fee = 0.09
   # local grid fee to be used
   local_grid_fee = 0
   # factor describing the relation of grid fee to cumulative power network edge weights
   weight_factor = 0.03

> Entering a grid fee matrix into a simulation is essential for accurate cost calculations related to using the grid. Ensure proper format, identifiers, units, and values for the rate matrix, whether complete or created from config file entries.

Reviewer: This description should stand in the beginning, e.g. after line 52
